Packages/com.unity.inputsystem/InputSystem/Editor/AssetEditor/InputActionAssetManager.cs

```diff
@@ -33,12 +41,16 @@ public string path
             }
         }
 
+        /// <summary>
+        /// Returns the name of the associated imported asset.
+        /// </summary>
         public string name
         {
             get
             {
-                if (m_ImportedAssetObject != null)
-                    return m_ImportedAssetObject.name;
+                var asset = importedAsset;
```

Comment:
looks like we could skip creating var asset here.

Reply:
I guess you are right, its kind of intentional though since importedAsset when not just cached is a costly operation. Hence I keep a local to avoid any side-effects or cost introduced by calling importedAsset multiple times. Since there are many access to asset variable in this getter though and importedAsset is non-trivial I rather just keep the local. Otherwise if someone is inventive down the line with importedAsset this getter could be causing problems.